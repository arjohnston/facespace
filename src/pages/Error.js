import React from 'react'
import { Link } from 'react-router-dom'

export default class Error extends React.Component {
  render () {
    return (
      <div className='container'>
        <h1>
          Whoops, looks like the page doesn't exist. <Link to='/'>Return home</Link>
        </h1>
      </div>
    )
  }
}
