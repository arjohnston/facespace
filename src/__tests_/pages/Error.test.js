/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Error from '../../pages/Error'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('With Snapshot Testing', () => {
  it('should match its empty snapshot', () => {
    const component = renderer.create(
      <Error />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('With Enzyme Testing', () => {
  it('renders without crashing', () => {
    shallow(<Error />)
  })
})
