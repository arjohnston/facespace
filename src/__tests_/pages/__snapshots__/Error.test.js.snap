// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should match its empty snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Whoops, looks like the page doesn't exist. 
    <Link
      to="/"
    >
      Return home
    </Link>
  </h1>
</div>
`;
